app/commands/runtimes/FixExerciseConfigs.php

```diff
@@ -0,0 +1,283 @@
+                    return false;
+                }
+
+                $pipeline['name'] = $newId;
+                $changes[$oldId] = $newId;
+            }
+            unset($pipeline); // just to make sure a reference is not accidentaly used
+        }
+        unset($test); // just to make sure a reference is not accidentaly used
+
+        return $changes;
+    }
+
+    /*
+     * Finally, the main function of command!
+     */
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        // just to save time (we do not have to pass this down to every other method invoked)
+        $this->input = $input;
+        $this->output = $output;
+
+        $this->nonInteractive = $input->getOption('yes');
+        $this->silent = $input->getOption('silent');
+        $runtimeId = $input->getArgument('runtime');
+
+        if (!$this->confirm("Attempting to fix exercise configurations for $runtimeId. Do you wish to proceed? ")) {
+            return Command::SUCCESS;
+        }
+
+        try {
+            $this->exerciseConfigs->beginTransaction();
+
+            // make sure internal pipeline caches are populated
+            $this->loadPipelines($runtimeId);
+
+            // take all configs one by one regardles of their exercises
+            $configs = $this->exerciseConfigs->findAll();
+            $updated = $failed = 0;
+            foreach ($configs as $configEntity) {
+                $configId = $configEntity->getId();
+                $config = $configEntity->getParsedConfig();
+                if (!in_array($runtimeId, $config['environments'] ?? []) || empty($config['tests'])) {
+                    continue;
+                }
+
+                // try to fix the config and get a list of changes...
+                $changes = $this->fixTestsPipelines($config['tests'], $runtimeId);
+
+                if ($changes === false) {
+                    // error occured
+                    $this->writeln("Cannot fix exercise config $configId, pipelines cannot be matched.");
+                    ++$failed;
+                } elseif ($changes) {
+                    // some changes are present -> save updated config and report
+                    $configEntity->overrideConfig($config);
+                    $this->exerciseConfigs->persist($configEntity);
+
+                    $this->writeln("Exercise config $configId has been fixed:");
+                    foreach ($changes as $oldId => $newId) {
+                        $this->writeln("\t$oldId => $newId");
+                    }
+                    ++$updated;
+                }
+            }
+
+            $this->exerciseConfigs->commit();
+            $this->writeln("Total $updated configs were updated, $failed configs are still possibly broken.");
+        } catch (Exception $e) {
+            $this->exerciseConfigs->rollback();
+            $msg = $e->getMessage();
+            $output->writeln("Error: $msg");
+            return 1;
```

Comment:
It seems a bit weird that there is `Command::SUCCESS`, but nothing for failures...